underfs/s3a/src/test/java/alluxio/underfs/s3a/S3AUnderFileSystemFactoryTest.java

```diff
@@ -23,19 +29,39 @@
  * Unit tests for the {@link S3AUnderFileSystemFactory}.
  */
 public class S3AUnderFileSystemFactoryTest {
+  private String mPath = "s3a://test-bucket/path";
+  private AlluxioConfiguration mAlluxioConf = ConfigurationTestUtils.defaults();
+  private UnderFileSystemConfiguration mConf = UnderFileSystemConfiguration.defaults(mAlluxioConf);
+  private UnderFileSystemFactory mFactory1 = UnderFileSystemFactoryRegistry.find(
+          "s3a://test-bucket/path", mAlluxioConf);
+  private UnderFileSystemFactory mFactory2 = UnderFileSystemFactoryRegistry.find(
+          "s3://test-bucket/path", mAlluxioConf);
+  private UnderFileSystemFactory mFactory3 = UnderFileSystemFactoryRegistry.find(
+          "s3n://test-bucket/path", mAlluxioConf);
 
   @Test
   public void factory() {
-    AlluxioConfiguration conf = ConfigurationTestUtils.defaults();
-    UnderFileSystemFactory factory = UnderFileSystemFactoryRegistry.find("s3a://test-bucket/path",
-        conf);
-    UnderFileSystemFactory factory2 = UnderFileSystemFactoryRegistry.find("s3://test-bucket/path",
-        conf);
-    UnderFileSystemFactory factory3 = UnderFileSystemFactoryRegistry.find("s3n://test-bucket/path",
-        conf);
-
-    Assert.assertNotNull(factory);
-    Assert.assertNotNull(factory2);
-    Assert.assertNull(factory3);
+    assertNotNull(mFactory1);
+    assertNotNull(mFactory2);
+    assertNull(mFactory3);
+  }
+
+  @Test
+  public void createInstanceWithNullPath() {
+    Exception e = Assert.assertThrows(NullPointerException.class, () -> mFactory1.create(
+            null, mConf));
+    assertTrue(e.getMessage().equals("path should not be null"));
+  }
+
+  @Test
+  public void createInstanceWithPath() {
+    UnderFileSystem ufs = mFactory1.create(mPath, mConf);
+    assertNotNull(ufs);
+    assertTrue(ufs instanceof S3AUnderFileSystem);
+  }
+
+  @Test
+  public void supportsValidPath() {
+    assertTrue(mFactory1.supportsPath(mPath));
```

Comment:
If you need to test an object, the object should be created in the test. Also you can write more tests verifying some paths are supported and others are Not.